fontc_crater/src/ci.rs

```diff
@@ -140,11 +132,54 @@ fn run_crater_and_save_results(args: &CiArgs) -> Result<(), Error> {
     // we write the map of target -> source repo to a separate file because
     // otherwise we're basically duplicating it for each run.
     let sources_file = args.out_dir.join(SOURCES_FILE);
-    super::try_write_json(&results.source_repos, &sources_file)
+    super::try_write_json(&source_repos, &sources_file)
 }
 
 fn result_path_for_current_date() -> String {
     let now = chrono::Utc::now();
     let timestamp = now.format("%Y-%m-%d-%H%M%S");
     format!("{timestamp}.json")
 }
+
+fn make_targets(cache_dir: &Path, repos: &[RepoInfo]) -> (Vec<Target>, BTreeMap<PathBuf, String>) {
+    let mut targets = Vec::new();
+    let mut repo_list = BTreeMap::new();
+    for repo in repos {
+        let Ok(iter) = repo.iter_configs(cache_dir) else {
+            log::warn!(
+                "error reading repo '{}'",
+                repo.repo_path(cache_dir).display()
+            );
+            continue;
+        };
+        for config_path in iter {
+            let config = match Config::load(&config_path) {
+                Ok(x) => x,
+                Err(e) => {
+                    log::warn!("failed to load config '{}': '{e}'", config_path.display());
+                    continue;
+                }
+            };
+            for source in &config.sources {
+                let src_path = config_path
+                    .parent()
+                    .expect("config path always in sources dir")
```

Comment:
nit: here and below we don't actually verify what dir, should it just be `config path is a file in a directory`?

Reply:
I use `expect` as basically an alternative to an assert, or an unwrap + comment.

These are both trying to communicate the invariant that the paths returns from `iter_configs` always start with the `cache_dir` argument, and necessarily include the `source` (or `Source`) directory component.